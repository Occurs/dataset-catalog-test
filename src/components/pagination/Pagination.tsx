import React from 'react';
import Typography from '@material-ui/core/Typography';
import Box from '@material-ui/core/Box';
import Button from '@material-ui/core/Button';

type TPagination = {
  totalPageCount: number;
  page: number;
  setPage: (page: number) => void;
};

export const Pagination = ({ totalPageCount, page, setPage }: TPagination): JSX.Element => {
  const onPageChangeHandler = (page: number) => () => {
    setPage(page);
  };
  return (
    <Box display="flex" alignItems="center" justifyContent="space-around" width="70%" margin="24px auto 0 auto">
      <Button size="small" color="primary" disabled={page === 1} onClick={onPageChangeHandler(1)}>
        First
      </Button>
      <Button size="small" color="primary" disabled={page === 1} onClick={onPageChangeHandler(page - 1)}>
        Previous
      </Button>
      <Typography>
        <Box component="span" margin="0 12px" fontSize="12px" width="100%">
          {`Page ${page} of ${totalPageCount}`}
        </Box>
      </Typography>
      <Button size="small" color="primary" disabled={page === totalPageCount} onClick={onPageChangeHandler(page + 1)}>
        Next
      </Button>
      <Button
        size="small"
        color="primary"
        disabled={page === totalPageCount}
        onClick={onPageChangeHandler(totalPageCount)}
      >
        Last
      </Button>
    </Box>
  );
};
