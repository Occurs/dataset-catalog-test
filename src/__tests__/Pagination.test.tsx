import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { Pagination } from '../components/pagination/Pagination';

describe('Paginator Component', () => {
  const props = {
    totalPageCount: 100,
    page: 5,
    setPage: jest.fn(),
  };

  test('snapshot', () => {
    const component = renderer.create(<Pagination {...props} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('render', () => {
    const { getByText } = render(<Pagination {...props} />);
    getByText('Page 5 of 100');
  });

  test('render with small itemsLength', () => {
    const propsForCurrentTest = {
      ...props,
      totalPageCount: 8,
    };
    const { getByText } = render(<Pagination {...propsForCurrentTest} />);
    getByText('Page 5 of 8');
  });

  test('render last page', async () => {
    const { getByText } = render(<Pagination {...props} page={7} />);
    getByText('Page 7 of 100');
  });

  test('click on previous page button', () => {
    const { getByText } = render(<Pagination {...props} />);
    const previous = getByText('Previous');
    fireEvent.click(previous);
    expect(props.setPage).toHaveBeenCalledTimes(1);
    expect(props.setPage).toHaveBeenCalledWith(4);
  });

  test('click on next page button', () => {
    const { getByText } = render(<Pagination {...props} />);
    const next = getByText('Next');
    fireEvent.click(next);
    expect(props.setPage).toHaveBeenCalledTimes(2);
    expect(props.setPage).toHaveBeenCalledWith(6);
  });

  test('click on first page button', () => {
    const { getByText } = render(<Pagination {...props} />);
    const first = getByText('First');
    fireEvent.click(first);
    expect(props.setPage).toHaveBeenCalledTimes(3);
    expect(props.setPage).toHaveBeenCalledWith(1);
  });

  test('click on last page button', () => {
    const { getByText } = render(<Pagination {...props} />);
    const last = getByText('Last');
    fireEvent.click(last);
    expect(props.setPage).toHaveBeenCalledTimes(4);
    expect(props.setPage).toHaveBeenCalledWith(100);
  });
});
