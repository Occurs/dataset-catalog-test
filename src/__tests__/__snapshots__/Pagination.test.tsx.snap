// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator Component snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-1"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      First
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Previous
    </span>
  </button>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    <span
      className="MuiBox-root MuiBox-root-2"
    >
      Page 5 of 100
    </span>
  </p>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Next
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-textSizeSmall MuiButton-sizeSmall"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Last
    </span>
  </button>
</div>
`;
